File: OneCodeTeam/How-to-use-WCF-to-implement-Reverse-Ajax-in-ASPNET-24728/[C#]-How to use WCF to implement Reverse Ajax in ASP.NET/C#/CSWCFReverseAJAX/Dispatcher.svc.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Runtime.Serialization;
using System.ServiceModel;
using System.ServiceModel.Activation;
using System.ServiceModel.Web;
using System.Text;

namespace CSWCFReverseAJAX
{
    [ServiceContract(Namespace = "")]
    [AspNetCompatibilityRequirements(RequirementsMode = AspNetCompatibilityRequirementsMode.Allowed)]
    public class Dispatcher
    {
        // To use HTTP GET, add [WebGet] attribute. (Default ResponseFormat is WebMessageFormat.Json)
        // To create an operation that returns XML,
        //     add [WebGet(ResponseFormat=WebMessageFormat.Xml)],
        //     and include the following line in the operation body:
        //         WebOperationContext.Current.OutgoingResponse.ContentType = "text/xml";

        /// <summary>
        /// Dispatch the new message event.
        /// </summary>
        /// <param name="userName">The loged in user name</param>
        /// <returns>the message content</returns>        
        [OperationContract]
        //[WebInvoke(Method = "POST", BodyStyle = WebMessageBodyStyle.Wrapped, ResponseFormat = WebMessageFormat.Json)]   
        [WebInvoke(RequestFormat = WebMessageFormat.Json, ResponseFormat = WebMessageFormat.Json)]
        public string WaitMessage(string userName)
        {
            return ClientAdapter.Instance.GetMessage(userName);
        }
    }
}

File: OneCodeTeam/How-to-use-WCF-to-implement-Reverse-Ajax-in-ASPNET-24865/[VB]-How to use WCF to implement Reverse Ajax in ASP.NET/VB/VBWCFReverseAJAX/Dispatcher.svc.vb
Imports System.Collections.Generic
Imports System.Linq
Imports System.Runtime.Serialization
Imports System.ServiceModel
Imports System.ServiceModel.Activation
Imports System.ServiceModel.Web
Imports System.Text


<ServiceContract([Namespace]:="")> _
<AspNetCompatibilityRequirements(RequirementsMode:=AspNetCompatibilityRequirementsMode.Allowed)> _
Public Class Dispatcher
    ' To use HTTP GET, add [WebGet] attribute. (Default ResponseFormat is WebMessageFormat.Json)
    ' To create an operation that returns XML,
    '     add [WebGet(ResponseFormat=WebMessageFormat.Xml)],
    '     and include the following line in the operation body:
    '         WebOperationContext.Current.OutgoingResponse.ContentType = "text/xml";

    ''' <summary>
    ''' Dispatch the new message event.
    ''' </summary>
    ''' <param name="userName">The loged in user name</param>
    ''' <returns>the message content</returns>
    ''' <remarks></remarks>
    '''[WebInvoke(Method = "POST", BodyStyle = WebMessageBodyStyle.Wrapped, ResponseFormat = WebMessageFormat.Json)]   
    <OperationContract> _
    <WebInvoke(RequestFormat:=WebMessageFormat.Json, ResponseFormat:=WebMessageFormat.Json)> _
    Public Function WaitMessage(userName As String) As String
        Return ClientAdapter.Instance.GetMessage(userName)
    End Function
End Class

